src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -910,6 +910,18 @@ public LiveReplicationPerformanceCollector.ReplicationPerformanceType GetReplica
 
         public void Dispose()
         {
+            if (_incomingWork != PoolOfThreads.LongRunningWork.Current)
+            {
+                try
+                {
+                    _incomingWork?.Join(int.MaxValue);
+                }
+                catch (ThreadStateException)
+                {
+                    // expected if the thread hasn't been started yet
+                }
+            }
+            _incomingWork = null;
```

Comment:
this should come after `_disposeOnce.Dispose();`, no?

Reply:
Done